dali/operators/image/crop/bbox_crop.cc

```diff
@@ -14,143 +14,635 @@
+      ", expected: ", ndim));
+
+    std::vector<int64_t> tmp(ndim);
+    for (int sample = 0; sample < batch_size; sample++) {
+      const auto* sample_data = arg_view.tensor_data(sample);
+      for (int d = 0; d < ndim; d++) {
+        tmp[d] = static_cast<int64_t>(sample_data[d]);
+      }
+      v.emplace_back(tmp);
+    }
+  } else if (spec.HasArgument(name)) {
+    auto tmp = spec.GetRepeatedArgument<int>(name);
+    DALI_ENFORCE(static_cast<int>(tmp.size()) == ndim,
+      make_string("Argument `", name, "` must be a ", ndim, "D vector"));
+
+    TensorShape<> sh(std::vector<int64_t>(tmp.begin(), tmp.end()));
+    v.resize(batch_size, sh);
+  } else {
+    if (required)
+      DALI_FAIL(make_string("Argument `", name, "` is required"));
+  }
+}
+
+struct SampleOption {
+  bool no_crop = false;
+  float min_iou = 0.0f;
+};
+
+struct Range {
+  bool Contains(float k) const {
+    assert(min <= max);
+    return k >= min && k <= max;
+  }
+  float min = 0.0f, max = 0.0f;
+};
+
+}  // namespace
+
 DALI_SCHEMA(RandomBBoxCrop)
     .DocStr(
-        R"code(Perform a prospective crop to an image while keeping bounding boxes and labels consistent. Inputs must be supplied as
-two Tensors: `BBoxes` containing bounding boxes represented as `[l,t,r,b]` or `[x,y,w,h]`, and `Labels` containing the
-corresponding label for each bounding box. Resulting prospective crop is provided as two Tensors: `Begin` containing the starting
-coordinates for the `crop` in `(x,y)` format, and 'Size' containing the dimensions of the `crop` in `(w,h)` format.
-Bounding boxes are provided as a `(m*4)` Tensor, where each bounding box is represented as `[l,t,r,b]` or `[x,y,w,h]`. Resulting
-labels match the boxes that remain, after being discarded with respect to the minimum accepted intersection threshold.
-Be advised, when `allow_no_crop` is `false` and `thresholds` does not contain `0` it is good to increase `num_attempts` as otherwise
-it may loop for a very long time.)code")
-    .NumInput(2)
-    .NumOutput(4)
+        R"code(Performs a prospective random crop to an image coordinate space while keeping
+bounding boxes (and optionally labels) consistent. That is, after applying the random crop operator
+to the image coordinate space, the bounding boxes will be adjusted or filtered out to match the
+cropped region of interest. The applied random crop operator is constrained by the arguments
+provided to the operator.
+
+The cropping window random selection is selected at random until it matches the overlap restrictions
+specified by `thresholds` argument. Thresholds values represent minimum intersection-over-union
+(IoU) of the cropping window and each of the provided bounding boxes. A valid crop is the one where
+all the provided bounding boxes produce an IoU above a give value present in `thresholds` randomly
+selected. Alternative, we can allow not to crop as one of the valid outcomes of the random process
```

Comment:
```suggestion
Alternatively, we can allow not to crop as one of the valid outcomes of the random process
```
(assuming that the previous line is adjusted according to the suggestion)